frontend/src/concepts/pipelines/content/PipelineAndVersionContext.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import { getPipelineIdByPipelineVersion } from '~/concepts/pipelines/content/utils';
+import { PipelineKF, PipelineVersionKF } from '~/concepts/pipelines/kfTypes';
+
+type PipelineAndVersionContextType = {
+  pipelineFactory: () => {
+    selectedPipelines: PipelineKF[];
+    setSelectedPipelines: React.Dispatch<React.SetStateAction<PipelineKF[]>>;
+  };
+  versionFactory: (pipelineId: string) => {
+    selectedVersions: PipelineVersionKF[];
+    setSelectedVersions: React.Dispatch<React.SetStateAction<PipelineVersionKF[]>>;
+  };
+  getResourcesForDeletion: () => { pipelines: PipelineKF[]; versions: PipelineVersionKF[] };
+  clearAfterDeletion: () => void;
+  isPipelineChecked: (pipelineId: string) => boolean;
+};
+
+type PipelineAndVersionContextProviderProps = {
+  children: React.ReactNode;
+};
+
+export const PipelineAndVersionContext = React.createContext<PipelineAndVersionContextType>({
+  pipelineFactory: () => ({ selectedPipelines: [], setSelectedPipelines: () => undefined }),
+  versionFactory: () => ({ selectedVersions: [], setSelectedVersions: () => undefined }),
+  getResourcesForDeletion: () => ({ pipelines: [], versions: [] }),
+  clearAfterDeletion: () => undefined,
+  isPipelineChecked: () => false,
+});
+
+const PipelineAndVersionContextProvider: React.FC<PipelineAndVersionContextProviderProps> = ({
+  children,
+}) => {
+  const [selectedPipelines, setSelectedPipelines] = React.useState<PipelineKF[]>([]);
+  const [selectedVersions, setSelectedVersions] = React.useState<{
+    [pipelineId: string]: PipelineVersionKF[];
+  }>({});
+
+  const setVersions =
+    (pipelineId: string) =>
+    (
+      newVersions: PipelineVersionKF[] | ((prevState: PipelineVersionKF[]) => PipelineVersionKF[]),
+    ) => {
+      setSelectedVersions((prev) => ({
+        ...prev,
+        [pipelineId]:
+          typeof newVersions === 'function' ? newVersions(prev[pipelineId] || []) : newVersions,
+      }));
+    };
+
+  const isPipelineChecked = (pipelineId: string) =>
+    selectedPipelines.map((pipeline) => pipeline.id).includes(pipelineId);
+
+  return (
+    <PipelineAndVersionContext.Provider
+      value={{
+        pipelineFactory: () => ({
```

Comment:
Still not sure factory was the best word lol... maybe `selector`? `pipelineDataSelector` and `versionDataSelector` ? naming is hard...